src/db/postgres/lrsql/postgres/record.clj

```diff
@@ -224,4 +231,54 @@
   (-query-platform-frequency [_ tx]
     (query-platform-frequency tx))
   (-query-timeline [_ tx input]
-    (query-timeline tx input)))
+    (query-timeline tx input))
+
+  bp/ReactionBackend
+  (-insert-reaction! [_ tx params]
+    (try
+      (insert-reaction! tx params)
+      (catch PSQLException ex
+        (if (= "23505" (.getSQLState ex))
+          :lrsql.reaction/title-conflict-error
+          (throw ex)))))
```

Comment:
Is this the canonical way to properly identify different PSQLExceptions?

Reply:
I'll defer to @cliffcaseyyet the PG expert on this one lol, but yeah I think this is the most reliable way to identify a conflict error rn

Reply:
@kelvinqian00 @milt Yes, it is pretty shit but we have precedent and explored other options. The codes are at times the ONLY identifiable characteristics of the exceptions. The good news is i believe they do not reuse codes so it should be permanently correct at best OR eventually irrelevant at worst.

Reply:
Ok I just wanted to note that calling `.getSQLState` is a lot better than having to use regexes to parse the error message.

Reply:
'tis